File: Competitive Programming 3/Section 1.3.3/11332 - Summing Digits/Python/main.py
def sumDigits(x):
    ans = 0 
    while x>0:
        ans+=x%10
        x//=10
    return ans
def main():
    while True:
        n = int(input())
        if n==0:
            break
        while n > 9 :
            n = sumDigits(n)
        print(n)
main()

File: Competitive Programming 3/Section 1.3.3/11332 - Summing Digits/C++/main.cpp
#include <iostream>

using namespace std;

int sumDigits(int x){
    int ans = 0;
    for(;x;x/=10){
        ans+=x%10;
    }
    return ans;
}

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    int n ; 
    while(cin >> n,n){ 
        while(n>9){
            n = sumDigits(n);
        }
        cout << n << "\n";
    }
    return 0;
}